dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -327,4 +330,32 @@ object TypedEncoder {
     recordEncoder: Lazy[RecordEncoderFields[G]],
     classTag: ClassTag[F]
   ): TypedEncoder[F] = new RecordEncoder[F, G]
+
+  type UDT[A >: Null] = FramelessInternals.PublicUserDefinedType[A]
+
+  /**
+    * Encodes things using a Spark SQL's User Defined Type (UDT) if there is one defined in implicit.
+    *
+    * Example: to use Spark ML's VectorUDT implementation as a TypedEncoder, add the following in your scope:
+    * {{{
+    * import org.apache.spark.ml.linalg.{Vector, SQLDataTypes}
+    * import frameless.TypedEncoder.UDT
+    * implicit val mLVectorUDT: UDT[Vector] = SQLDataTypes.VectorType.asInstanceOf[UDT[Vector]]
```

Comment:
@kanterov yes indeed, a spark-ml user could do that to access directly the private org.apache.spark.ml.linalg.VectorUDT and avoid downcasting. The issue if we want to declare these implicits UDTs directly in Frameless is that frameless-dataset will need a dependency on spark-mllib... which I don't think we want to impose.